go/vt/vttablet/tabletserver/repltracker/writer.go

```diff
@@ -298,9 +342,13 @@ func (w *heartbeatWriter) killWrite() error {
 }
 
 // allowNextHeartbeatRequest ensures that the next call to RequestHeartbeats() passes through and
-// is not dropped.
+// is not rate-limited.
 func (w *heartbeatWriter) allowNextHeartbeatRequest() {
-	atomic.AddInt64(&w.onDemandRequestTicks, 1)
+	// The writer could be close()d while this function is running and thus the value of the rate limiter could be nil.
+	// We thus use golang atomic here to avoid locking mutexes.
+	if rateLimiter := w.safeOnDemandRequestsRateLimiter(); rateLimiter != nil {
+		rateLimiter.AllowOne()
+	}
```

Comment:
IMO using the atomic.Pointer value here (assuming there aren't good reasons to switch to that type) would be clearer and safer too.

Reply:
Done!